packages/ui-reactive/src/Unlocking.tsx

```diff
@@ -36,6 +36,29 @@ export class UnlockingDisplay extends React.PureComponent<Props> {
     : null ;
   }
 
+  private groupByEra (list: UnlockChunk[]) {
+    let eraMap: Map<string, Balance> = new Map<string, Balance>();
```

Comment:
Difficult since you return a map. A reduce on a object would have worked.

```js
return list.reduce((map, { era, value }) => {
  const key = era.toString();

  if (!map[key)) {
    map[key] = value;
  } else {
    map[key] = map[key].add(value);
  }

  return map;
}, {} as { [index: string]: BN });
```

Reply:
thanks for the guidance I'll give it another shot